core/arch/riscv64/mangle.c

```diff
@@ -0,0 +1,219 @@
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL RIVOS, INC. OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+ * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+ * DAMAGE.
+ */
+
+#include "../globals.h"
+#include "arch.h"
+
+void
+mangle_arch_init(void)
+{
+    /* FIXME i#3544: Not implemented */
+    ASSERT_NOT_IMPLEMENTED(false);
+}
+
+void
+insert_clear_eflags(dcontext_t *dcontext, clean_call_info_t *cci, instrlist_t *ilist,
+                    instr_t *instr)
+{
+    /* FIXME i#3544: Not implemented */
+    ASSERT_NOT_IMPLEMENTED(false);
+}
+
+uint
+insert_push_all_registers(dcontext_t *dcontext, clean_call_info_t *cci,
+                          instrlist_t *ilist, instr_t *instr, uint alignment,
+                          opnd_t push_pc,
+                          reg_id_t scratch /*optional*/
+                              _IF_AARCH64(bool out_of_line))
+{
+    /* FIXME i#3544: Not implemented */
+    ASSERT_NOT_IMPLEMENTED(false);
+    return 0;
+}
+
+void
+insert_pop_all_registers(dcontext_t *dcontext, clean_call_info_t *cci, instrlist_t *ilist,
+                         instr_t *instr, uint alignment _IF_AARCH64(bool out_of_line))
+{
+    /* FIXME i#3544: Not implemented */
+    ASSERT_NOT_IMPLEMENTED(false);
+}
+
+reg_id_t
+shrink_reg_for_param(reg_id_t regular, opnd_t arg)
+{
+    /* FIXME i#3544: Not implemented */
+    ASSERT_NOT_IMPLEMENTED(false);
+    return regular;
+}
+
+uint
+insert_parameter_preparation(dcontext_t *dcontext, instrlist_t *ilist, instr_t *instr,
+                             bool clean_call, uint num_args, opnd_t *args)
+{
+    /* FIXME i#3544: Not implemented */
+    ASSERT_NOT_IMPLEMENTED(false);
+    return 0;
+}
+
+bool
+insert_reachable_cti(dcontext_t *dcontext, instrlist_t *ilist, instr_t *where,
+                     byte *encode_pc, byte *target, bool jmp, bool returns, bool precise,
+                     reg_id_t scratch, instr_t **inlined_tgt_instr)
+{
+    /* FIXME i#3544: Not implemented */
+    ASSERT_NOT_IMPLEMENTED(false);
+    return false;
+}
+
+int
+insert_out_of_line_context_switch(dcontext_t *dcontext, instrlist_t *ilist,
+                                  instr_t *instr, bool save, byte *encode_pc)
+{
+    /* FIXME i#3544: Not implemented */
+    ASSERT_NOT_IMPLEMENTED(false);
+    return false;
+}
+
+/*###########################################################################
+ *###########################################################################
+ *
+ *   M A N G L I N G   R O U T I N E S
```

Comment:
Modify to exclude the extra spaces between characters. So that it fits in with other DR code's style.

Reply:
Done. I've copied this straight from x86 and aarchxx versions of this file. Those will also need aligning I guess.

Reply:
We can optionally fix those old files in this PR, but I'm fine either way.

Reply:
Done. Better do it now than forget in the future. :)

Reply:
Thanks!